src/peergos/shared/user/UserContext.java

```diff
@@ -1374,15 +1374,19 @@ private static void ensureAllowedToShare(FileWrapper file, String ourname, boole
         });
     }
 
-    public CompletableFuture<Boolean> shareWriteAccessWithAll(Path fileToShare,
-                                                               Set<String> writersToAdd) {
+
+    @JsMethod
+    public CompletableFuture<Boolean> shareWriteAccessWith(String pathToFile,
```

Comment:
In general a typed api is better than a string api. So I'd prefer if the JS method called the typed api. I'm hoping at somepoint we can remove all these JS only apis into a sub class or something. Java code for sure should be using the typed api.